### ISSUE TITLE
5 node local cluster does not work when connected to corporate VPN

### ISSUE THREAD

**Issue by framgeld** [microsoft/service-fabric#1005]
Service Fabric Runtime Version: 7.1.428.9590

Environment: Dev cluster

Description:
Unable to use a 5-node local cluster when connected to the corporate VPN. Due to "work at home" policies, it is imperative that I can get this working over the corporate VPN.

Observed behavior:
With the corporate VPN disconnected, SF Manager will start cluster, says it connect to it, and eventually create 5 instances of FabricApplicationGateway.exe.
When connected to corporate VPN, SF Manager will start cluster but is unable to connect to it. A single instance of FabricApplicationGateway.exe intermittently appears then disappears.

Expected Behavior:
5-node local cluster is expected to work whether I am connected to a VPN or not.

OS: Windows (10.0.18363.959)

Assignees: @microsoft/service-fabric-triage

**Comment by framgeld** [comment]
Everything works as expected when using a 1-node local cluster.

**Comment by maburlik** [comment]
Are you creating the nodes with address "localhost" or an IP/FQDN? Try variations of this since some NIC interfaces may not be able to accommodate end-to-end communication through VPN.

**Comment by framgeld** [comment]
I am using the default method that Service Fabric uses to create the local dev cluster (no tweaks). I find it odd that Service Fabric works as expected with the VPN active when a 1-node cluster is created, but not when a 5-node cluster is created. I will, however, investigate how I change the setting(s) that you have referred to.

**Comment by framgeld** [comment]
Thanks to the pointer from @maburlik, I was able to create a 5-node cluster that has the FQDN of "localhost" instead of my full computer name (inclusive of domain name).

In the directory "C:\Program Files\Microsoft SDKs\Service Fabric\ClusterSetup\NonSecure\FiveNode", I changed the "iPAddress" setting from "ComputerFullName" to "localhost".

Service Fabric Local Cluster Manager thinks it failed to create the cluster though... but that is a minor annoyance rather than a blocking issue.
